db-engine/src/main/java/com/splicemachine/db/iapi/sql/dictionary/StatementSchemaPermission.java

```diff
@@ -128,6 +151,87 @@ public void check( LanguageConnectionContext lcc,
+            // check if a permission exists for the current role (if set).
+            String role = lcc.getCurrentRoleId(activation);
+
+            if (role != null) {
+
+                // Check that role is still granted to current user or
+                // to PUBLIC: A revoked role which is current for this
+                // session, is lazily set to none when it is attempted
+                // used.
+                String dbo = dd.getAuthorizationDatabaseOwner();
+                RoleGrantDescriptor rd = dd.getRoleGrantDescriptor
+                        (role, currentUserId, dbo);
+
+                if (rd == null) {
+                    rd = dd.getRoleGrantDescriptor(
+                            role,
+                            Authorizer.PUBLIC_AUTHORIZATION_ID,
+                            dbo);
+                }
+
+                if (rd == null) {
+                    // We have lost the right to set this role, so we can't
+                    // make use of any permission granted to it or its
+                    // ancestors.
+                    lcc.setCurrentRole(activation, null);
+                } else {
+                    // The current role is OK, so we can make use of
+                    // any permission granted to it.
+                    //
+                    // Look at the current role and, if necessary, the
+                    // transitive closure of roles granted to current role to
+                    // see if permission has been granted to any of the
+                    // applicable roles.
+
+                    RoleClosureIterator rci =
+                            dd.createRoleClosureIterator
+                                    (activation.getTransactionController(),
+                                            role, true /* inverse relation*/);
+
+                    String r;
+
+                    while ((authorization != AUTHORIZED) && (r = rci.next()) != null) {
+                            authorization = oneAuthHasPermissionOnSchema
+                                    (dd, r, forGrant);
+                    }
+
+                    if (authorization == AUTHORIZED) {
+                        // Also add a dependency on the role (qua provider), so
+                        // that if role is no longer available to the current
+                        // user (e.g. grant is revoked, role is dropped,
+                        // another role has been set), we are able to
+                        // invalidate the ps or activation (the latter is used
+                        // if the current role changes).
+                        DependencyManager dm = dd.getDependencyManager();
+                        RoleGrantDescriptor rgd =
+                                dd.getRoleDefinitionDescriptor(role);
+                        ContextManager cm = lcc.getContextManager();
+
+                        dm.addDependency(ps, rgd, cm);
+                        dm.addDependency(activation, rgd, cm);
```

Comment:
The distributed part of dependency management is not functioning it looks like for roles...
https://splicemachine.atlassian.net/browse/DB-6670

Reply:
Thanks John for bringing this up to my attention. I'll take a look at DB-6670. But I think this is a general role grant/revoke issue, not specific to the modify schema privilege in this ticket, we can proceed with the review of this PR, right?